src/main/java/rx/internal/operators/OperatorConcatEmptyWith.java

```diff
@@ -0,0 +1,193 @@
+        private final EmptyProducer emptyProducer;
+        private final Observable<? extends R> alternate;
+
+        ParentSubscriber(Subscriber<? super R> child, final SerialSubscription ssub, Observable<? extends R> alternate) {
+            this.child = child;
+            this.ssub = ssub;
+            this.emptyProducer = new EmptyProducer();
+            this.alternate = alternate;
+        }
+
+        @Override
+        public void setProducer(final Producer producer) {
+            /*Nothing from this producer will ever be requested as we never expect any items to be emitted from the
+            parent. The only thing w.r.t backpressure that is required is to store the requested count from the
+            child till the time the alternate observable is subscribed. So, this producer is just ignored and the
+            configured EmptyProducer on the child stores any buffered requested items till subscription to alternate.*/
+        }
+
+        @Override
+        public void onCompleted() {
+            if (!child.isUnsubscribed()) {
+                AlternateSubscriber as = new AlternateSubscriber(child, emptyProducer);
+                ssub.set(as);
+                alternate.unsafeSubscribe(as);
+            }
+        }
+
+        @Override
+        public void onError(Throwable e) {
+            child.onError(e);
+        }
+
+        @Override
+        public void onNext(T t) {
+            onError(new IllegalStateException("Concat empty with source emitted an item: " + t));
+        }
+    }
+
+    private final class AlternateSubscriber extends Subscriber<R> {
+
+        private final EmptyProducer emptyProducer;
+        private final Subscriber<? super R> child;
+
+        AlternateSubscriber(Subscriber<? super R> child, EmptyProducer emptyProducer) {
+            this.child = child;
+            this.emptyProducer = emptyProducer;
+        }
+
+        @Override
+        public void setProducer(final Producer producer) {
+            emptyProducer.setAltProducer(producer);
+        }
+
+        @Override
+        public void onCompleted() {
+            child.onCompleted();
+        }
+
+        @Override
+        public void onError(Throwable e) {
+            child.onError(e);
+        }
+
+        @Override
+        public void onNext(R r) {
+            child.onNext(r);
+        }
+    }
+
+    /**
+     * This is a producer implementation that only does the following:
+     * <ul>
+     <li>If the alternate producer has not yet arrived, store the total requested count from downstream.</li>
+     <li>If the alternate producer has arrived, then relay the request demand to it.</li>
+     </ul>
+     *
+     * Since, this is only applicable to this operator, it does not check for emissions from the source, as the source
+     * is never expected to emit any item. Thus it is "lighter" weight than {@link ProducerArbiter}
+     */
+    private static final class EmptyProducer implements Producer {
```

Comment:
I started off with `ProducerArbiter` and then thought that the functionality in `ProducerArbiter` is much larger than what is required here primarily because of the fact that the first producer here would never ever emit an item and hence the use case that `ProducerArbiter` is solving of resuming from an unsatisfied request from downstream, is a potential overhead. I would love to benchmark if there is a difference at all (although by sheer LOC it seems like an overhead) with this approach.

I have put this reasoning behind this class in the javadoc as a disclaimer :)